File: app/src/main/java/it/emanuelemelini/copypasta/http/APIInterface.java
package it.emanuelemelini.copypasta.http;

import it.emanuelemelini.copypasta.http.login.LoginResponseObj;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface APIInterface {

    @FormUrlEncoded
    @POST("login")
    Call<LoginResponseObj> login(@Field("username") String username,
                                    @Field("password") String password);

}

File: app/src/main/java/it/emanuelemelini/copypasta/http/APIInterface.kt
package it.emanuelemelini.copypasta.http

import it.emanuelemelini.copypasta.http.login.BaseLoginResponse
import it.emanuelemelini.copypasta.http.login.User
import it.emanuelemelini.copypasta.model.LoginModel
import retrofit2.Call
import retrofit2.http.*

interface APIInterface {

    @FormUrlEncoded
    @POST("login")
    fun loginOld(
        @Header("Content-Type") content: String = "application/json",
        @Field("IDuser") IDuser: Int = 0,
        @Field("username") username: String,
        @Field("password") password: String,
        @Field("IDgroup") IDgroup: Int = 0,
        @Field("groupAdmin") groupAdmin: Boolean = false,
        @Field("admin") admin: Boolean = false,
        @Field("deleted") deleted: Boolean = false
    ): Call<BaseResponse<User>>

    @POST("login")
    fun login(
        @Header("Content-Type") content: String = "application/json",
        @Body loginModel: LoginModel
    ): Call<BaseLoginResponse>

}